pkg/schemadsl/parser/parser.go

```diff
@@ -157,6 +157,40 @@ func (p *sourceParser) consumeTypeReference() AstNode {
 	return refNode
 }
 
+// consumeSpecificType consumes an identifier as a specific type reference, with optional caveat.
+func (p *sourceParser) consumeSpecificTypeWithCaveat() AstNode {
```

Comment:
is the definition of the caveat not yet part of the parser or did I miss something?

Reply:
Correct; that's in the next PR. This only adds the ability to *use* a caveat on a relation